### PR TITLE
Enable to set timezone Calendar for writing and reading timestamp data

### PR DESCRIPTION
**Pull request by hokorn**

### CODE REVIEW

File: src/main/java/os/failsafe/executor/FailsafeExecutor.java [hunk 1]
```diff
@@ -14,12 +14,7 @@
 import java.time.Duration;
 import java.time.LocalDateTime;
 import java.time.temporal.ChronoUnit;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-import java.util.Set;
-import java.util.UUID;
+import java.util.*;
```
Comment (cinemast) [inline src/main/java/os/failsafe/executor/FailsafeExecutor.java:17]: please revert and avoid `*` imports

File: src/main/java/os/failsafe/executor/TaskRepository.java [hunk 2]
```diff
@@ -7,16 +7,9 @@
 
 import java.io.IOException;
 import java.io.StringReader;
-import java.sql.Connection;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Statement;
-import java.sql.Timestamp;
+import java.sql.*;
```
Comment (cinemast) [inline src/main/java/os/failsafe/executor/TaskRepository.java:10]: please revert and avoid `*` imports

File: src/main/java/os/failsafe/executor/TaskRepository.java [hunk 3]
```diff
@@ -7,16 +7,9 @@
...
 import java.time.LocalDateTime;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
Comment (cinemast) [inline src/main/java/os/failsafe/executor/TaskRepository.java:12]: please revert and avoid `*` imports

File: src/main/java/os/failsafe/executor/utils/Database.java [hunk 4]
```diff
@@ -2,13 +2,11 @@
 
 import javax.sql.DataSource;
 import java.io.StringReader;
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Statement;
+import java.sql.*;
```
Comment (cinemast) [inline src/main/java/os/failsafe/executor/utils/Database.java:5]: please revert and avoid `*` imports

**Review by cinemast — CHANGES_REQUESTED** [review]

**Comment by oliverselinger** [comment]
Just had a look at the code. Couldn't this be resolved by passing a custom implementation of the SystemClock to the FailsafeExecutor constructor instead of the DefaultSystemClock which uses LocalDateTime provider?

The only timestamp that is not set by the system clock as far as I can see, is the creationTime. But this can also be set via SystemClock. So there wouldn't be a need for the TimeZone parameter.

**Comment by hokorn** [comment]
> Just had a look at the code. Couldn't this be resolved by passing a custom implementation of the SystemClock to the FailsafeExecutor constructor instead of the DefaultSystemClock which uses LocalDateTime provider?
> 
> The only timestamp that is not set by the system clock as far as I can see, is the creationTime. But this can also be set via SystemClock. So there wouldn't be a need for the TimeZone parameter.

I will try

File: src/main/java/os/failsafe/executor/TaskRepository.java (hunk 2 above)
Comment (hokorn) [reply src/main/java/os/failsafe/executor/TaskRepository.java]: Fixed

File: src/main/java/os/failsafe/executor/FailsafeExecutor.java (hunk 1 above)
Comment (hokorn) [reply src/main/java/os/failsafe/executor/FailsafeExecutor.java]: Fixed

**Pull request closed without merging**